scientific_library/tvb/simulator/models/deco2020_transcriptional.py

```diff
@@ -0,0 +1,323 @@
+        domain=Range(lo=0.0, hi=1.0, step=0.01),
+        doc="""Inhibitory global coupling scaling""")
+
+    state_variable_range = Final(
+        default={
+            "S_e": numpy.array([0.0, 1.0]),
+            "S_i": numpy.array([0.0, 1.0]),
+            "H_e": numpy.array([0.0, 100]),
+            "I_e": numpy.array([0.0, 30.0])
+        },
+        label="State variable ranges [lo, hi]",
+        doc="Population firing rate")
+
+    # Used for phase-plane axis ranges and to bound random initial() conditions.
+    state_variable_boundaries = Final(
+        label="State Variable boundaries [lo, hi]",
+        default={"S_e": numpy.array([0.0, 1.0]),
+                 "S_i": numpy.array([0.0, 1.0]),
+                 "H_e": numpy.array([0.0, 150.0]),
+                 "I_e": numpy.array([0.0, 30.0])},
+        doc="""The values for each state-variable should be set to encompass
+            the boundaries of the dynamic range of that state-variable. Set None for one-sided boundaries""")
+
+    variables_of_interest = List(
+        of=str,
+        label="Variables watched by Monitors",
+        choices=('H_e', 'I_e', 'S_i', 'S_e'),
+        default=('H_e', 'I_e', 'S_i', 'S_e'),
+        doc="""default state variables to be monitored""")
+
+    state_variables = ['S_e', 'S_i', 'H_e', 'I_e']
+    _nvar = 4
+    cvar = numpy.array([0], dtype=numpy.int32)
+
+    def configure(self):
+        """  """
+        super(Deco2020Transcriptional, self).configure()
+        self.update_derived_parameters()
+
+    def initial(self, dt, history_shape, rng=numpy.random):
+        """Generates uniformly distributed initial conditions,
+        bounded by the state variable limits defined by the model.
+        """
+        ic = numpy.empty(history_shape)
+
+        ic[:, 0] = 0.001
+        ic[:, 1] = 0.001
+        ic[:, 2] = 0.001
+        ic[:, 3] = 0.001
+
+        return ic
+
+    def _numpy_dfun(self, state_variables, coupling, local_coupling=0.0):
+        r"""
+        """
+
+        S = state_variables[:, :]
+
+        S_e = S[:, 0]
+        S_i = S[:, 1]
+
+        c_0 = coupling[:, 0]
+
+        # if applicable
+        lc_0 = local_coupling * S_e
+
+        coupling = self.G * self.J_N * (c_0 + lc_0)
+
+        J_N_S_e = self.J_N * S_e
+
+        I_ext = 0.0  # Resting state
+
+        inh = self.J_i * S_i
+
+        I_e = self.W_e * self.I_o + self.w_p * J_N_S_e + coupling - inh + I_ext
+
+        gain = 1.0 + self.alpha + self.beta * self.ratio
+        x_e = (self.a_e * I_e - self.b_e) * gain
+        H_e = x_e / (1 - numpy.exp(-self.d_e * x_e))
+
+        dS_e = - (S_e / self.tau_e) + (1.0 - S_e) * H_e * self.gamma_e
+
+        I_i = self.W_i * self.I_o + J_N_S_e - S_i + self.lamda * coupling
+
+        x_i = (self.a_i * I_i - self.b_i) * gain
+        H_i = x_i / (1 - numpy.exp(-self.d_i * x_i))
+
+        dS_i = - (S_i / self.tau_i) + H_i * self.gamma_i
+
+        dH_e = H_e - S[:, 2]
+        dI_e = I_e - S[:, 3]
```

Comment:
these state variables don't do anything, if you want to monitor intermediate variables during integrations, please see https://groups.google.com/g/tvb-users/c/ZcTkZaGQa10/m/SjBnEA71AQAJ